test/unit/clients/adminusers_client/service/update_merchant_name_test.js

```diff
@@ -0,0 +1,72 @@
+'use strict'
+
+// NPM dependencies
+const Pact = require('pact')
+const chai = require('chai')
+const chaiAsPromised = require('chai-as-promised')
+
+// Custom dependencies
+const path = require('path')
+const PactInteractionBuilder = require('../../../../fixtures/pact_interaction_builder').PactInteractionBuilder
+const getAdminUsersClient = require('../../../../../app/services/clients/adminusers_client')
+const serviceFixtures = require('../../../../fixtures/service_fixtures')
+
+// Constants
+const SERVICE_RESOURCE = '/v1/api/services'
+const port = Math.floor(Math.random() * 48127) + 1024
+const adminUsersClient = getAdminUsersClient({ baseUrl: `http://localhost:${port}` })
+const expect = chai.expect
+
+// Global setup
+chai.use(chaiAsPromised)
+
+describe('adminusers client - patch request to update merchant name', function () {
+  let provider = Pact({
+    consumer: 'selfservice',
+    provider: 'adminusers',
+    port: port,
+    log: path.resolve(process.cwd(), 'logs', 'mockserver-integration.log'),
+    dir: path.resolve(process.cwd(), 'pacts'),
+    spec: 2,
+    pactfileWriteMode: 'merge'
+  })
+
+  before(() => provider.setup())
+  after(done => provider.finalize().then(done()))
+
+  describe('a valid update merchant name patch request', () => {
+    const serviceExternalId = 'cp5wa'
+    const serviceName = 'serviceName'
+    const validUpdateMerchantNameRequest = serviceFixtures.validUpdateMerchantNameRequest('updated-merchant-details-name')
+    const validUpdateMerchantNameResponse = serviceFixtures.validUpdateMerchantNameResponse({
+      external_id: serviceExternalId,
+      name: serviceName
```

Comment:
the response should use the `validServiceResponse` fixture and include:
```
merchant_details: {
    name: 'updated-merchant-details-name'
}
````

It looks like currently it works due to the default values in the fixture you've added, but any field we are asserting on should be explicitly passed into the fixture as we shouldn't rely on the fixture defaults.

Reply:
Thanks, I have updated this to:
`merchant_details: {
        name: opts.value || ''
      }`

I have amended a test, now on line 121, to additionally assert on when a pre-filled organisation name is updated prior to submitting the form.